tests/test.linter.js

```diff
@@ -884,6 +884,47 @@ describe('Linter.extractMetadata()', function() {
     });
   });
 
+    it('should not scan known JS libraries', function() {
+    var addonLinter = new Linter({ _: ['foo'] });
+    var markJSFilesSpy = sinon.spy(addonLinter, '_markJSLibs');
+    addonLinter.checkFileExists = fakeCheckFileExists;
+    addonLinter.scan = () => Promise.resolve();
+    // suppress output.
+    addonLinter.print = sinon.stub();
+
+    var fakeFiles = {
+      'my/nested/library/path/j.js': 'jquery-2.1.4.min.js',
+    };
+
+    class FakeXpi {
+      getFile(path) {
+        return Promise.resolve(
+          fs.readFileSync(`tests/fixtures/jslibs/${fakeFiles[path]}`));
+      }
+      getFiles() {
+        var files = {};
+        for (let filename of Object.keys(fakeFiles)) {
+          files[filename] = { uncompressedSize: 5 };
+        }
+        return Promise.resolve(files);
+      }
+      getFilesByExt() {
+        return Promise.resolve(Object.keys(fakeFiles));
+      }
+    }
+
+    return addonLinter.extractMetadata({
+      _console: fakeConsole,
+      _Xpi: FakeXpi,
+    }).then((metadata) => {
+      var warnings = addonLinter.collector.warnings;
+      var filtered = warnings.map((element) => {
```

Comment:
This is a good test but a comment above it, just to explain to someone a bit less familiar with the code, would be great.